### PR TITLE
Version golangci-lint and build it with make

### PR DESCRIPTION
**Pull request by Kubuxu**
`golangci-lint` and `no-map-range` are in a separate go.mod file so they do not increase dependency surface of specs-actors

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,16 +18,20 @@ tidy:
...
 gen:
 	$(GO_BIN) run ./gen/gen.go
 .PHONY: gen
+
+
+# tools
+toolspath:=support/tools
+
+$(toolspath)/golangci-lint: $(toolspath)/go.mod
+	(cd $(toolspath); go build github.com/golangci/golangci-lint/cmd/golangci-lint)
```
Comment (Stebalien) [inline Makefile:30]:
```suggestion
	(cd $(toolspath); go build -o "$@" github.com/golangci/golangci-lint/cmd/golangci-lint)
```

Comment (Stebalien) [reply Makefile:30]: (just to be safe)

File: .circleci/config.yml [hunk 2]
```diff
@@ -93,24 +93,11 @@ jobs:
...
-              cd $(mktemp -d);
-              go mod init a;
-              go get github.com/golangci/golangci-lint/cmd/golangci-lint@v<< parameters.golangci-lint-version >>
-              go build github.com/golangci/golangci-lint/cmd/golangci-lint
-              mkdir -p $HOME/.local/bin
-              mv -f golangci-lint $HOME/.local/bin/
-            )
-      - run:
-          command: make .nomaprange.so
+          command: make -j3 support/tools/golangci-lint support/tools/no-map-range.so
```
Comment (Stebalien) [inline .circleci/config.yml:96]: `-j$(nproc)`?

File: .gitignore [hunk 3]
```diff
@@ -15,3 +15,6 @@
 # vendor/
 
 .idea
+
+support/tools/golangci-lint
+support/tools/no-map-range.so
```
Comment (Stebalien) [inline .gitignore:20]: Note: It's probably better to just have a `support/tools/bin` directory.

File: Makefile (hunk 1 above)
Comment (Kubuxu) [reply Makefile:30]: `$@` is relative, and we have to `cd` for go.mod to work

File: .circleci/config.yml (hunk 2 above)
Comment (Kubuxu) [reply .circleci/config.yml:96]: I think circle reports way more cores than it really gives you.

Comment (Stebalien) [reply .circleci/config.yml:96]: Fair enough.

File: Makefile (hunk 1 above)
Comment (Kubuxu) [reply Makefile:30]: Did this: `$(@:$(toolspath)/%=%)`

**Pull request merged**